tests/TestCase/ORM/Behavior/TranslateBehaviorTest.php

```diff
@@ -93,7 +93,6 @@ public function testCustomTranslationTable()
 
         $this->assertSame('CustomI18n', $i18n->getName());
         $this->assertInstanceOf(CustomI18nTable::class, $i18n->getTarget());
-        $this->assertSame('test_custom_i18n_datasource', $i18n->getTarget()->getConnection()->configName());
```

Comment:
Associations can't use different connections. This isn't the right place to test custom default connection name for Table class.

Reply:
Associations can use different connections, however they need to use the `select` strategy and not join/subquery.